### DISCUSSION TITLE
Public schema vs private schema - Why both?

### DISCUSSION THREAD

**Discussion by Vadorequest** [supabase/supabase#617] [Questions] [answered]
I'm not familiar with Postgre schemas.

Why do you use both a public and a private schema for users? What are the pros/cons?

> Even though Supabase provides an auth.users table, it is helpful to also create a users table in the public schema, which uses the same UID Primary Key as the auth.users. For security purposes, the auth schema is not exposed on the auto-generated API. Created a public.users table allows you to interact via the Supabase client - especially useful for cross-table queries.
>
> https://supabase.io/docs/guides/auth#create-a-publicusers-table

I don't understand your motivations, despite reading the documentation.

**Comment by shorn** [comment, marked as answer]
TL:DR;
Data in the  `public` schema is "open by default", data in other schemas is "restricted by default".

----

By default, tables in the `public` schema can be read/written by clients via the Supabase REST API.  That means anybody can access table data  in the `public` schema if they have your API key (usually referenced in the doco as the [supabase key](https://supabase.io/docs/client/initializing#parameters)).  The API key is usually [shared via your client application](https://github.com/supabase/supabase/discussions/256), you must presume everyone in the world knows your API key because there's no practical way to keep it secret.

If you want to restrict read or write access to tables in the public schema - you need to enable [row level security](https://supabase.io/docs/guides/auth#row-level-security) for each table and define policies declaring who is allowed to read or write to that table.

Tables in any other schema are not directly accessible via the REST API.  If you want non-public data to be accessible - you need to create an access mechanism in the `public` schema (a view, or a function, etc.)

The contents of the `auth` schema are sensitive - you don't want that data to be accessible to anyone in the world who has access to your API key.

Please note: I am not part of the Supabase team - just trying to help.

**Reply by Vadorequest** [reply]
Thank you, it's much more clearer!

> Data in the public schema is "open by default", data in other schemas is "restricted by default".

This sentence should be in the documentation 😂  💯
